### ISSUE TITLE
[Bug]: accidentally ending a title with a `/` breaks storybook

### ISSUE THREAD

**Issue by cellog** [storybookjs/storybook#26642]
### Describe the bug

with a title like:

```ts
  title: "scenes/MessageComposition/shared/ComposerImageLibraryPanel/",
```

Storybook fails to compile

### To Reproduce

create a story with this title:
```ts
  title: "scenes/MessageComposition/shared/ComposerImageLibraryPanel/",
```

### System

```bash
Storybook Environment Info:

  System:
    OS: macOS 14.4
    CPU: (10) arm64 Apple M1 Pro
    Shell: 5.9 - /bin/zsh
  Binaries:
    Node: 20.11.1 - ~/.nvm/versions/node/v20.11.1/bin/node
    Yarn: 1.22.19 - ~/.nvm/versions/node/v20.11.1/bin/yarn
    npm: 10.2.4 - ~/.nvm/versions/node/v20.11.1/bin/npm <----- active
  Browsers:
    Chrome: 123.0.6312.58
    Safari: 17.4
```

### Additional context

![image](https://github.com/storybookjs/storybook/assets/98250/27f23d15-9044-48f6-992f-9f1ced1aa36d)

**Comment by hmcostaa** [comment]
@vanessayuenn I would like to take this issue. I just have some doubts about what is the optimal solution in this scenario. Do we allow the creation of the story by checking if there is nothing after the "/" or we contain the error to that story and allow the rest of the stories to be rendered?

**Comment by shilman** [comment]
@hmcostaa  The current behavior is mostly correct. Having a `/` at the end of the `title` property is not valid CSF, and will break indexing. However, if you click on the error the message that it shows is hard to parse.

<img width="847" alt="image" src="https://github.com/storybookjs/storybook/assets/488689/14612d11-c478-4766-adc6-c890be2d2b3b">

Full text:

> Error: Invalid part '', leading to id === parentId ('example-button'), inside title 'Example/Button/'

I think it should show a more useful error like: 

> Invalid title `Example/Button/` ending in slash.

I could imagine a different UI that shows the rest of the stories and ALSO surfaces the story errors. But that's a much bigger change.
